### ISSUE TITLE
Replace rusoto with aws-sdk-rust

### ISSUE THREAD

**Issue by DaniPopes** [gakonst/ethers-rs#2312]
We should replace the unmaintained `rusoto` crates with the official and actively maintained `aws-sdk-rust`.

References:
- #2190
- [RUSTSEC-2022-0071](https://rustsec.org/advisories/RUSTSEC-2022-0071)]

**Comment by prestwich** [comment]
agree. this will technically be a break change due to the ability to pass in a client

**Comment by prestwich** [comment]
i looked at this, and i am on the fence. the aws sdk is still in preview, and is not recommended for prod. Absent a stable thing to update to, I'm not sure we should move off the thing that is currently working

**Comment by iajoiner** [comment]
@prestwich rusoto is abandonware and has been so for more than a year. We do in fact need to switch.

**Comment by yorhodes** [comment]
running into this quite consistently with the aws kms signer https://github.com/rusoto/rusoto/issues/1686
